// @serverless/platform-sdk/dist/deployments/updateDeployment.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

function _asyncToGenerator(fn) { return function () { var gen = fn.apply(this, arguments); return new Promise(function (resolve, reject) { function step(key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { return Promise.resolve(value).then(function (value) { step("next", value); }, function (err) { step("throw", err); }); } } return step("next"); }); }; }

var fetch = require('isomorphic-fetch');
var platformConfig = require('../config');
var currentVersion = require('../../package.json').version;

var updateDeployment = function () {
  var _ref = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee(data) {
    var body, response, text;
    return regeneratorRuntime.wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            body = {
              deployment: {
                status: data.status
              }
            };

            if (data.state && data.state.service && data.state.service.name) {
              body.state = data.state;
            }
            _context.next = 4;
            return fetch(`${platformConfig.backendUrl}tenants/${data.tenant}/applications/${data.app}/services/${data.serviceName}/deployments/${data.deploymentId}`, {
              method: 'PUT',
              body: JSON.stringify(body),
              headers: {
                'Content-Type': 'application/json',
                'x-platform-version': currentVersion,
                Authorization: `bearer ${data.accessKey}`
              }
            });

          case 4:
            response = _context.sent;

            if (response.ok) {
              _context.next = 10;
              break;
            }

            _context.next = 8;
            return response.text();

          case 8:
            text = _context.sent;
            throw new Error(text);

          case 10:
            return _context.abrupt('return', response.json());

          case 11:
          case 'end':
            return _context.stop();
        }
      }
    }, _callee, undefined);
  }));

  return function updateDeployment(_x) {
    return _ref.apply(this, arguments);
  };
}();

exports.default = updateDeployment;
